scripts/list_hardware.py

```diff
@@ -9,26 +9,31 @@
 from dataclasses import dataclass
 from pathlib import Path, PurePath
 
-import pykwalify.core
+import jsonschema
 import yaml
+from jsonschema.exceptions import best_match
 
 try:
     from yaml import CSafeLoader as SafeLoader
 except ImportError:
     from yaml import SafeLoader
 
 
-SOC_SCHEMA_PATH = str(Path(__file__).parent / 'schemas' / 'soc-schema.yml')
+SOC_SCHEMA_PATH = str(Path(__file__).parent / 'schemas' / 'soc-schema.yaml')
 with open(SOC_SCHEMA_PATH) as f:
-    soc_schema = yaml.load(f.read(), Loader=SafeLoader)
+    soc_schema = yaml.load(f, Loader=SafeLoader)
```

Comment:
Does it work without reading first?

```suggestion
    soc_schema = yaml.load(f.read(), Loader=SafeLoader)
```

Reply:
:| Sorry...